x-pack/plugins/lens/public/editor_frame_service/editor_frame/expression_helpers.ts

```diff
@@ -114,22 +115,33 @@ export function buildExpression({
       state: unknown;
     }
   >;
-  framePublicAPI: FramePublicAPI;
+  framePublicAPI: {
+    datasourceLayers: Record<string, DatasourcePublicAPI>;
+    query: Query;
+    dateRange?: DateRange;
+    filters: Filter[];
+  };
```

Comment:
I should have expanded on that in the description - this change is necessary because when building the expression in the embeddable factory, we don't have a meaningful date range available. This change (together with the very first one you commented on) is relaxing the requirement to provide a date range in those places.